src/dsp_tools/utils/xmlupload/stash_circular_references.py

```diff
@@ -18,122 +24,92 @@
 logger = get_logger(__name__)
 
 
-def _stash_circular_references(
-    nok_resources: list[XMLResource],
-    ok_res_ids: set[str],
-) -> tuple[list[XMLResource], set[str], list[XMLResource], Stash | None]:
+def stash_circular_references(
+    resources: list[XMLResource],
+    stash_lookup: dict[str, list[str]],
+) -> Stash | None:
     """
-    Raises:
-        BaseError
+    Stashes problematic resource-references from a list of resources.
+    The resources are modified in-place.
+
+    Args:
+        resources: all resources of the XML file
+        stash_lookup: A dictionary which maps the resources that have stashes to the UUIDs of the stashed links
+
+    Returns:
+        stash: an object that contains the stashed references
     """
     stashed_standoff_values: list[StandoffStashItem] = []
     stashed_link_values: list[LinkValueStashItem] = []
-    ok_resources: list[XMLResource] = []
 
-    for res in nok_resources.copy():
+    for res in resources.copy():
+        if not res.id in stash_lookup:
+            continue
         for link_prop in res.get_props_with_links():
             if link_prop.valtype == "text":
                 for value in link_prop.values:
-                    if value.resrefs and not all(_id in ok_res_ids for _id in value.resrefs):
-                        # replace the problematic XML with a UUID
-                        # and remove the problematic resrefs from the XMLValue's resrefs list
-                        standoff_xml = cast(KnoraStandoffXml, value.value)
-                        uuid = str(uuid4())
-                        standoff_stash_item = StandoffStashItem(
-                            res_id=res.id,
-                            res_type=res.restype,
-                            uuid=uuid,
-                            prop_name=link_prop.name,
-                            value=standoff_xml,
-                        )
-                        stashed_standoff_values.append(standoff_stash_item)
-                        value.value = KnoraStandoffXml(uuid)
-                        value.resrefs = [_id for _id in value.resrefs if _id in ok_res_ids]
+                    if not value.link_uuid in stash_lookup[res.id]:
```

Comment:
I think it's recommended to write `x not in y` rather than `not x in y` but I may be mistaken